### ISSUE TITLE
trio_websocket makes wrong assumptions about `trio.__version__`

### ISSUE THREAD

**Issue by mgorny** [HyperionGray/trio-websocket#178]
```pytb
$ python3.11 -c 'import trio_websocket'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.11/site-packages/trio_websocket/__init__.py", line 1, in <module>
    from ._impl import (
  File "/usr/lib/python3.11/site-packages/trio_websocket/_impl.py", line 33, in <module>
    _TRIO_MULTI_ERROR = tuple(map(int, trio.__version__.split('.'))) < (0, 22, 0)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: '0+dev'
```

Apparently it assumes that `trio.__version__` will always be X.Y.Z and doesn't support other valid PEP 440 versions, effectively causing it to fail when a development version of trio is used (but the same would also happen with release candidates, etc.).

**Comment by belm0** [comment]
Thank you.  Rather than add some parsing library dependency or large regex, I think looking at the first two numbers will cover likely cases.

`tuple(map(int, trio.__version__.split('.')[:2])) < (0, 22)`

**Comment by mgorny** [comment]
Yes, I think that should be good enough. Thanks!

**Comment by belm0** [comment]
the fix is published to pypi (v0.10.3)
